api/v1/service/resource/resource.go

```diff
@@ -518,3 +521,45 @@ func (s *service) GetRawYamlByCatalogKindNameVersion(ctx context.Context, p *res
 
 	return io.NopCloser(bytes.NewBuffer(content)), nil
 }
+
+// Fetch a raw resource yaml file using the name of catalog, resource name, and kind
+func (s *service) GetLatestRawYamlByCatalogKindName(ctx context.Context, p *resource.GetLatestRawYamlByCatalogKindNamePayload) (io.ReadCloser, error) {
+	s.Logger(ctx).Info(fmt.Sprintf("Fetching Latest YAML for resource %s", p.Name))
+
+	version, err := getLatestVersion(s.DB(ctx), p.Name, p.Kind, p.Catalog)
+
+	if err != nil {
+		return nil, err
+	}
+
+	yamlPath := fmt.Sprintf("%s/%s/%s/%s/%s", s.CatalogClonePath(), strings.ToLower(p.Catalog), strings.ToLower(p.Kind), p.Name, version)
+	yamlPath = fmt.Sprintf("%s/%s.yaml", yamlPath, p.Name)
+
+	content, err := os.ReadFile(yamlPath)
+	if err != nil {
+		return nil, resource.MakeNotFound(fmt.Errorf("resource not found"))
+	}
+
+	return io.NopCloser(bytes.NewBuffer(content)), nil
+}
+
+// Fetch the latest verion of a resource
+func getLatestVersion(db *gorm.DB, name string, kind string, catalogName string) (string, error) {
+
+	var catalogInfo *model.Catalog
+	if err := db.Where(&model.Catalog{Name: catalogName}).Find(&catalogInfo).Error; err != nil {
+		return "", resource.MakeInternalError(err)
+	}
+
+	var resInfo *model.Resource
+	if err := db.Where(&model.Resource{Name: name, Kind: strings.ToUpper(kind), CatalogID: catalogInfo.ID}).Find(&resInfo).Error; err != nil {
+		return "", resource.MakeInternalError(err)
+	}
+
+	var allVersions []string
+
+	if err := db.Model(&model.ResourceVersion{}).Select("version").Where(&model.ResourceVersion{ResourceID: resInfo.ID}).Find(&allVersions).Error; err != nil {
```

Comment:
noob question: will this return the versions in sorted order?

Reply:
yes